CONTRIBUTING.md

```diff
@@ -271,23 +271,6 @@ re-generate the set of dependencies, run the following in the repository and com
 nox -s pip-compile
 ```
 
-## Releasing
```

Comment:
What do you think about moving this paragraph someplace else instead of just dropping it?

Reply:
Agreed. We can create a private repo to serve as storage for maintainer-only related stuff. Losing the information completely doesn't sound like a good idea to me.

Reply:
Does it need to be private though? Just a MAINTAINING.md should work -- if someone who is not a maintainer is willing to read it then by all means, there is not secret information there, but they are unlikely to gain any insight from that. Having more than one repo adds to maintenance burden, having all relevant docs in one place makes it easier to locate necessary data.

Reply:
Fair enough. I was trying to foresee more information in there that we may not want to share publicly, like the infra details, and maintenance information, etc. after all, we're currently not solely relying on GH's native infra....